Documentation/GettingStarted.md

```diff
@@ -1,104 +1,861 @@
+
+e.g.
+
+```swift
+public func mapOrDie<E, R>
+    (selector: E -> RxResult<R>)
+    -> (Observable<E> -> Observable<R>) {
+    return { source in
+        return selectOrDie(selector)(source)
+    }
+}
+
+public func map<E, R>
+    (selector: E -> R)
+        -> (Observable<E> -> Observable<R>) {
+    return { source in
+        return select(selector)(source)
+    }
+}
 ```
 
-If you have a `|>` operator defined as a pipe operator in your project, you can use it too instead of `>-` operator
+Returning an error from a selector will cause entire graph of dependent sequence transformers to "die" and fail with error. Dying implies that it will release all of its resources and never produce another sequence value. This is usually not an obvious effect.
+
+If there is some `UITextField` bound to a observable sequence that fails with error or completes, screen won't be updated ever again.
+
+To make those situations more obvious, RxCocoa debug build will throw an exception in case some sequence that is bound to UI control terminates with an error.
+
+Using functions without "OrDie" suffix is usually a more safe option.
+
+There is also the `catch` operator for easier error handling.
+
+## Debugging
+
+Using debugger alone is useful, but you can also use `>- debug`. `debug` operator will print out all events to standard output and you can add also label those events.
+
+`debug` acts like a probe. Here is an example of using it:
 
 ```swift
-let a /*: Observable<Int>*/ = Variable(1)
-let b /*: Observable<Int>*/ = Variable(2)
+let subscription = myInterval(0.1)
+    >- debug("my probe")
+    >- map { e in
+        return "This is simply \(e)"
+    }
+    >- subscribeNext { n in
+        println(n)
+    }
 
-// immediately prints: 3 is positive
-combineLatest(a, b) { $0 + $1 } 
-    |> filter { $0 >= 0 } 
-    |> map { "\($0) is positive" }
-    |> subscribeNext { println($0) }
+NSThread.sleepForTimeInterval(0.5)
+
+subscription.dispose()
 ```
 
-The choice is yours.
+will print
+
+```
+[my probe] subscribed
+Subscribed
+[my probe] -> Event Next(Box(0))
+This is simply 0
+[my probe] -> Event Next(Box(1))
+This is simply 1
+[my probe] -> Event Next(Box(2))
+This is simply 2
+[my probe] -> Event Next(Box(3))
+This is simply 3
+[my probe] -> Event Next(Box(4))
+This is simply 4
+[my probe] dispose
+Disposed
+```
+
+You can also use `subscribe` instead of `subscribeNext`
+
+```
+NSURLSession.sharedSession().rx_JSON(request)
+   >- map { json in
+       return parse()
+   }
+   >- subscribe { n in      // this subscribes on all events including error and completed
+       println(n)
+   }
+```
+
+## Variables
+
+`Variable`s represent some observable state. `Variable` without containing value can't exist because initializer requires initial value.
+
+Variable is a [`Subject`](http://reactivex.io/documentation/subject.html). More specifically a `BehaviorSubject`. The reason why `BehaviorSubject` as `Variable` alias is just for convenience.
```

Comment:
> More specifically a `BehaviorSubject`. The reason why `BehaviorSubject` as `Variable` alias is just for convenience. 

More specifically it is a `BehaviorSubject`. Because `BehaviorSubject` is aliased as `Variable`  for convenience.